guides/shipments/solidus-active-shipping-extension.md

```diff
@@ -0,0 +1,217 @@
+
+## Access pre-configured calculators
+
+The `solidus_active_shipping` extension comes with pre-configured shipping
+calculators. Administrators can access these calculators when adding new
+shipping methods by picking them from the "Base Calculator" drop-down menu.
+
+You can extend or override the pre-configured calculator classes, such as
+`Spree::Calculator::Shipping::Usps::GlobalExpressGuaranteed`.
+
+You can see all of the extension's included shipping calculators in the list
+of [the extension's shipping models][models].
+
+[models]: https://github.com/solidusio-contrib/solidus_active_shipping/tree/master/app/models/spree/calculator/shipping
+
+## Add additional shipping calculators
+
+Additional delivery services that are not pre-configured as a calculator in the
+extension can be easily added.
+
+### Inherit from an existing service's base class
+
+For example, you need to a delivery service called "Bogus First Class
+International" from USPS, you can add a new calculator class that inherits from
+the `Spree::Calculator::Shipping::Usps::Base` class:
+
+```ruby
+module Spree
+  module Calculator::Shipping
+    module Usps
+      class BogusFirstClassInternational < Spree::Calculator::Shipping::Usps::Base
+        self.description
+          "USPS Bogus First Class International"
+        end
+      end
+    end
+  end
+end
+```
+
+Unlike shipping calculators that you write yourself, these calculators inherit
+from the existing superclasses built into `solidus_active_shipping` and do not
+require a `compute` instance method that returns a shipping amount.
+
+### Match the delivery service's provided name
+
+The string returned by the `description` method must match the name of the USPS
+delivery service _exactly_. To determine the exact spelling, you should examine
+what the USPS API returns.
+
+The following code block returns the description and the rate of USPS delivery
+services:
+
+```ruby
+class Calculator::ActiveShipping < Calculator
+  def compute(line_items)
+    #....
+    rates = retrieve_rates(origin, destination, packages(order))
+    # the key of this hash is the name you need to match
+    # raise rates.inspect
+
+    return nil unless rates
+    rate = rates[self.description].to_f +
+(Spree::ActiveShipping::Config[:handling_fee].to_f || 0.0)
+    return nil unless rate
+    # divide by 100 since active_shipping rates are expressed as cents
+
+    return rate/100.0
+  end
+
+  def retrieve_rates(origin, destination, packages)
+    #....
+    # carrier is an instance of ActiveMerchant::Shipping::USPS
+    response = carrier.find_rates(origin, destination, packages)
+    # turn this beastly array into a nice little hash
+    h = Hash[*response.rates.collect { |rate| [rate.service_name, rate.price]
+}.flatten]
+    #....
+  end
+end
+```
+
+<!-- TODO:
+  Give more context to the code block above. Would you/how would you
+  practically use this in your app? -->
```

Comment:
Maybe just remove this section for now. `solidus_active_shipping` is a beast, and hard to customize or understand...

Reply:
I ended up removing most of the content under the "Match the delivery service's provided name", but do you mean this whole article?